meson.build

```diff
@@ -0,0 +1,80 @@
+project('gpaste',
+  'c',
+  version: '3.30.1',
+  meson_version: '>=0.46.0',
+)
+
+apiversion = '1.0'
+gpaste_soversion = '11.0.2'
+
+glib_req='>=2.58.0'
+gdk_pixbuf_req='>=2.38.0'
+gjs_req='>=1.54.0'
+gtk3_req='>=3.24.0'
+mutter_clutter_req='>=3.30.0'
+
+appstream_dep = dependency('appstream-glib')
+gdk_dep = dependency('gdk-3.0', version: gtk3_req)
+gdk_pixbuf_dep = dependency('gdk-pixbuf-2.0', version: gdk_pixbuf_req)
+gdk_x11_dep = dependency('gdk-x11-3.0', version: gtk3_req)
+gio_dep = dependency('gio-2.0', version: glib_req)
+glib_dep = dependency('glib-2.0', version: glib_req)
+gobject_dep = dependency('gobject-2.0', version: glib_req)
+gtk_dep = dependency('gtk+-3.0', version: gtk3_req)
+pango_dep = dependency('pango')
+x11_dep = dependency('x11')
+xi_dep = dependency('xi')
+
+dbus_service_dir = dependency('dbus-1').get_pkgconfig_variable('session_bus_services_dir')
+keybindings_dir = dependency('gnome-keybindings').get_pkgconfig_variable('keysdir')
+
+if get_option('introspection')
+  mutter_clutter_dep = dependency('mutter-clutter-3', version: mutter_clutter_req)
+  gjs_dep = dependency('gjs-1.0', version: gjs_req)
+endif
+
+if get_option('systemd')
+  unitdir=dependency('systemd').get_pkgconfig_variable('systemduserunitdir')
+endif
+
+if get_option('vapi') and not get_option('introspection')
+  error('GIR generation has to be enabled via -Dintrospection=true to build vala bindings!')
+endif
+
+add_project_arguments(
+  '-DG_PASTE_COMPILATION',
+  '-DDGTK_API=3.0',
```

Comment:
Since GPaste doesn't support GTK 4.0 via autotools yet, I've just hardcoded 3.0 for now.